client/src/components/DownloadModal.js

```diff
@@ -45,7 +51,17 @@ export const DownloadModal = ({
 
   return (
     <>
-      {icon ? (
+      {icon && sampleMetadataOnly ? (
+        <Anchor
+          icon={icon}
+          onClick={handleClick}
+          disabled={disabled}
+          label={
+            sampleMetadataOnly ? <Text color="brand">{buttonLabel}</Text> : ''
```

Comment:
Currently, the rendering of the metadata-only download button label ("Download Sample Metadata") depends on whether the button is for a metadata-only download, not the availability of a metadata download file.

If the metadata download file is available: 
![Screenshot 2024-06-13 at 1 44 01 PM](https://github.com/AlexsLemonade/scpca-portal/assets/31800566/cc691b0e-1ef1-4a8a-a7e4-1ce2a3f21d40)

if it's unavailable:
![Screenshot 2024-06-13 at 1 45 19 PM](https://github.com/AlexsLemonade/scpca-portal/assets/31800566/3bcb3dd8-a6ad-4df0-9c9d-c0081824ab6b)

Unless we hide the metadata-only download button when the metadata download file is not available?

Please let me know your input on this!

(See commit at 43a94f814d4d48c4445abb93cc2deea3da4a150f)

Reply:
That isn't quite the point I was trying to make.

In use modal this PR introduces:
```javascript
  const buttonLabel = `Download ${
    sampleMetadataOnly ? 'Sample Metadata' : resourceType
  }`
```

which is functionally equivalent to:
```javascript
  const buttonLabel = `Download ${
    initialPublicComputedFile.metadata_only ? 'Sample Metadata' : resourceType
  }`
```

I think however that this confusion illustrates that the current usage of the `DownloadModal` has drifted from the original plan of "give it a resource and let it figure out how to render itself". So, I think at this point we should make the following changes.

- In the `DownloadModal` component add the props: `color`, `label`. We will pass this in wherever we use the component instead of trying to evaluate it based on the resource or computedFile that is passed in.
- In `useDownloadModal` let's remove the variable `buttonTitle` and add to the `modalTitle` at the end if it is metadata.
- Delete the `MetadataDownloadModal` component, it really only serves as a place to call the hook that gets the metadata file and returns if the file is available. We can do that in `ProjectSearchResult` and `ProjectSamplesTable` components when determining how to render the button.
- Remove all arguments of `sampleMetadataOnly` for components and hooks, we want to determine state based off of the resource and the computedFile or initialComputedFile.

Reply:
The separation of the UI presentation logic from the `useDownloadModal` hook is a great idea! Since it eliminates the need for us to conditionally determine the button label's text (especially now that we have more than two resource types).

I applied your input in the following commits: 
> In the `DownloadModal` component add the props: `color`, `label`. We will pass this in wherever we use the component instead of trying to evaluate it based on the resource or `computedFile` that is passed in.

See the commit at c1c929b32cb6c1beaae3c838972ed7f8cd158270.

> In `useDownloadModal` let's remove the variable `buttonTitle` and add to the `modalTitle` at the end if it is metadata.

See the commits at 93068098cc14591b444d62b42ec42de3b67d139c and ffda1378cf771575e97643d3c10447d0fa9a8499.

**NOTE:** Instead of using `initialPublicComputedFile`, `publicComputedFie` is used to evaluate `isSampleMetadataOnly`, since after removing ‘buttonLabel’, this evaluation is only necessary for the downloading modal.

> Delete the `MetadataDownloadModal` component, it really only serves as a place to call the hook that gets the metadata file and returns if the file is available. We can do that in `ProjectSearchResult` and `ProjectSamplesTable` components when determining how to render the button.

See the commitd at dfd55f0a5bb8c2e687a685f620b8469a768971fb and 967baffa6d7e5986afdc65f6a69d0637667001d3.

**NOTE:** In this commit, I've also adjusted the Project Download button (in `ProjectHeader.js`), since `buttonLabel` was removed.

> Remove all arguments of `sampleMetadataOnly` for components and hooks, we want to determine state based off of the resource and the `computedFile` or `initialComputedFile`.

They were completed in above commits, and yes, I used `computedFie` 👍 !